src/orion/service/broker/local.py

```diff
@@ -0,0 +1,198 @@
+
+log = logging.getLogger(__name__)
+
+
+class LocalExperimentBroker(ExperimentBroker):
+    """Creates an experiment client for each request and process the request.
+    Parallel requests will instantiate different clients which might generate race conditions,
+    locks are required.
+
+    Everything keeps being reinstantiated.
+
+    This is easy to scale because nothing is kept between request,
+    we can just load balance the request through n servers.
+
+    """
+
+    def new_experiment(self, request: RequestContext) -> Dict:
+        log.debug("Spawning new experiment")
+
+        client = create_experiment_client(request)
+        # pylint: disable=protected-access
+        euid = str(client._experiment.id)
+
+        # pylint: disable=protected-access
+        return success(
+            dict(
+                experiment_name=str(client.name),
+                euid=euid,
+                space=client._experiment.space.configuration,
+                version=client._experiment.version,
+                mode="x",
+                max_trials=client._experiment.max_trials,
+                working_dir=client._experiment.working_dir,
+                metadata=client._experiment.metadata,
+            )
+        )
+
+    def suggest(self, request: RequestContext) -> Dict:
+
+        # NOTE: need to fix lost trials here
+        # so reschedule them
+        experiment_name = request.data.pop("experiment_name")
+        client = retrieve_experiment_client(request, experiment_name)
+
+        if client.is_broken:
+            raise BrokenExperiment()
+
+        trial = client.suggest(**request.data)
+
+        small_trial = self._make_trial(trial)
+        return success(dict(trials=[small_trial]))
+
+    def _make_trial(self, trial):
+        return RemoteTrial.make_json_from_trial(trial)
+
+    def observe(self, request: RequestContext) -> Dict:
+        experiment_name = request.data.pop("experiment_name")
+        client = retrieve_experiment_client(request, experiment_name)
+        storage = client.storage
+
+        trial_hash = request.data.pop("trial_hash")
+        results = request.data.get("results")
+
+        # pylint: disable=protected-access
+        trial = storage.get_trial(uid=trial_hash, experiment_uid=client._experiment.id)
+
+        assert trial is not None, "Trial not found"
+        client.observe(trial, results)
+
+        return success({})
+
+    def is_done(self, request: RequestContext) -> Dict:
+        experiment_name = request.data.pop("experiment_name")
+        client = retrieve_experiment_client(request, experiment_name)
+
+        if client.is_broken:
+            raise BrokenExperiment()
+
+        return success(dict(is_done=client.is_done))
+
+    def heartbeat(self, request: RequestContext) -> Dict:
+        storage = get_storage_for_user(request)
+        trial_id = request.data.get("trial_id")
+
+        # pylint: disable=protected-access
+        results = storage._db._db["trials"].update_one(
```

Comment:
Why not use storage.heartbeat? Is it because of the username? Shouldn't it be handled by this PR's modifications in MongoDB?